Module 2

PrivetAdel opened the pull request.

PrivetAdel committed:
build: добавить зависимости

добавила зависимости:
@testing-library/react
@material-ui/core
loft-taxi-mui-theme

prop-types

PrivetAdel committed:
style(components): начать приводить приложение в соответствие с дизайном

Начала приводить внешний вид страниц, созданных на предыдущем этапе в соответствие с макетом, с
использованием material-ui

PrivetAdel committed:
style(components): style(components): начать приводить приложение в соответствие с дизайном

Поправила стили и разбила форму на доп компоненты

PrivetAdel committed:
feat(components): используя context-api создать контекст авторизации

Используя context-api и хук useContext создала контекст авторизации

PrivetAdel committed:
feat(components): добавить карту mapbox на страницу с картой

Используя context-api и хук useContext создала контекст авторизации

PrivetAdel committed:
feat(components): указать типы пропсов для компонентов

Указала типы пропсов для компонентов с помощью PropTypes

PrivetAdel committed:
fix(components): добавить удаление карты | изменить провайдер контекста

Добавила удаление карты, перенесла логику в провайдер контекста

PrivetAdel committed:
test(components): добавить тестирование компонентов

Начала писать тесты

PrivetAdel committed:
fix(components): добавть файл setupTests.js

Добавила файл setupTests.js

PrivetAdel committed:
test(components): протестировать компоненты

Начала писать тесты для компонентов, не все тесты рабочие

bread-bread commented on src/App.test.js:
```diff
@@ -0,0 +1,9 @@
+import App from './App';
+
+test('render App component', () => {
+  render(<App />);
```
App нужно обернуть в Provider, так как внтури app используются данные из контекста авторизации

bread-bread commented on src/components/FormLogin.test.js:
```diff
@@ -0,0 +1,70 @@
+
+describe("FormLogin", () => {
+  const props = {
+    onSignUp: jest.fn, 
```
mock фунции инициализируются через вызов jest.fn()

bread-bread commented on src/components/FormLogin.test.js:
```diff
@@ -0,0 +1,70 @@
+
+    const link = getByText(/Регистрация/i);
+    fireEvent.click(link);
+    expect(link).toHaveBeenCalled();
```
здесь нужно проверять вызываемую mock функцию, т.е. `expect(props.onSignUp).toHaveBeenCalled()`

bread-bread commented on src/components/FormLogin.test.js:
```diff
@@ -0,0 +1,70 @@
+  });
+
+  it("click on submit button", () => {
+    const onSubmitHandler = jest.fn();
```
эту функцию переименуем в loginMock

bread-bread commented on src/components/FormLogin.test.js:
```diff
@@ -0,0 +1,70 @@
+
+    const {getByLabelText, getByRole} = render(
+      <AuthorizationContecxt.Provider value={{isLoggedIn}}>
+        <FormLogin {...props}>
```
Оставляем просто <FormLogin {...props} />, а в провайдер передаем в качестве login функции loginMock
`<AuthorizationContecxt.Provider value={{isLoggedIn, login: loginMock}}>`

bread-bread commented on src/components/FormLogin.test.js:
```diff
@@ -0,0 +1,70 @@
+    const inputPassword = getByLabelText(/Пароль/i);
+    fireEvent.change(inputPassword, {target: {value: '123123'}});
+
+    fireEvent.click(getByRole('button'));
```
вызываем fireEvent.submit(getByTestId('form'));

PrivetAdel committed:
fix(components): поправить тесты, поправить добавление карты

Поправила тесты для App, FormLogin, FormSignUp
Переустановила mapbox-gl версии 1.13.0
